File: frontend/src/stocks/Stock.ts
export class Stock {
    id: number | undefined;
    name: string = '';
    description: string = '';
    imageUrl: string = '';
    price: number = 0;
    availableUnits: number | undefined;
    isActive: boolean = false;
    amount: number = 0;

    constructor(initializer?: any) {
        if (!initializer) return;
        if (initializer.id) this.id = initializer.id;
        if (initializer.name) this.name = initializer.name;
        if (initializer.description) this.description = initializer.description;
        if (initializer.imageUrl) this.imageUrl = initializer.imageUrl;
        if (initializer.price) this.price = initializer.price;
        if (initializer.availableUnits) this.availableUnits = initializer.availableUnits;
        if (initializer.isActive) this.isActive = initializer.isActive;
        if (initializer.amount) this.amount = initializer.amount;
    }
}

File: stockservice/src/main/java/com/carlospatinos/stockservice/model/Stock.java
package com.carlospatinos.stockservice.model;

import java.io.Serializable;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.RequiredArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Getter
@Setter
@NoArgsConstructor
@RequiredArgsConstructor
@AllArgsConstructor
@ToString(includeFieldNames = false)
public class Stock implements Serializable {
    @lombok.NonNull
    private Integer id;
    @lombok.NonNull
    private String name;
    @lombok.NonNull
    private Double price;
    @lombok.NonNull
    private Integer availableUnits;

    private String description;
    private String imageUrl;
    private Integer amount;
}
